packages/runtime/src/parser/parser.ts

```diff
@@ -860,28 +860,19 @@ export function assertMaxCost(
 	}
 }
 
-// verify max length of TEAL code is within consensus parameters
-function _assertMaxLen(len: number, mode: ExecutionMode): void {
-	if (mode === ExecutionMode.SIGNATURE) {
+/**
+ * verify max size of lsig
+ * @param lsigProgramArgsSize lsig program and arugments size
+ * @param mode Execution mode - Signature (stateless) OR Application (stateful)
+ */
+function assertLogicMaxLen(lsigProgramArgsSize: number, mode: ExecutionMode): void {
+	if (mode === ExecutionMode.SIGNATURE && lsigProgramArgsSize > LogicSigMaxSize) {
```

Comment:
what does Args refers to?

Reply:
Do we check it both for args and length?

Reply:
args is arugments.
yes, we have to check for both args size and lsgi size.

Reply:
Please see here: https://github.com/algorand/go-algorand-sdk/blob/13da8508530b7dc2e9ae62766cccd0baaff0f4bd/logic/logic.go#L83

they have used the same logic.